src/components/Layout/Sidebar.tsx

```diff
@@ -47,13 +49,14 @@ function Sidebar() {
     );
   };
 
-  const renderMenuItem = (key: string, text: string, icon: string) => {
+  const renderMenuItem = (key: string, text: string, icon: string, additionalNav?: { icon: string, onClick: () => void }) => {
```

Comment:
I think it makes more sense to use the existing pattern and create a separate component, just like renderNonLinkMenuItem, when most of it is done then we can refactor into single more complex component.